src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -216,13 +234,25 @@ public void SetiOSKeychainSecurityGroup(string keychainSecurityGroup)
         public virtual void Clear()
         {
             AccessTokenCacheDictionary.Clear();
```

Comment:
Threading issue: Between Clear and setting CacheSize to zero, something may get added to the dictionary and the two can go out of sync

Reply:
True, but we also don't lock on cache operations in general (based on eventual consistency principle). So this behavior can happen with other add/remove methods.

Reply:
Eventual consistency is not guaranteed here. For example:
1. Thread 1 calls the AccessTokenCacheDictionary.Clear() method and is paused.
2. Thread 2 adds an item and updates the _appCacheSize.
3. Thread 1 resumes execution and sets the appCacheSize to 0 (Interlocked.Exchange(ref _appCacheSize, 0)).
4. Now there is one item in the cache but the size is 0.
5. The same issue applies to the TokenCache.CacheSize, and in this case the error can accumulate over multiple Clear() calls.

Reply:
@dannybtsai Do you suggest locking? Only when clearing or always when updating the cache size?

Reply:
It is hard to achieve consistency without locking as far as I can tell. And you need to always lock, not just when clearing cache otherwise it is the same problem.